### DISCUSSION TITLE
Show usd layers in maya without maya-usd

### DISCUSSION THREAD

**Discussion by vinRosso** [Autodesk/maya-usd#3998] [Q&A]
Hi maya-usd community, I'm here for your help.

While writing our new pipeline, we faced some problems with the maya-usd integration.
Especially, the maya limit of 2GB for included usd data, and the allocation in memory of anonymous layers that breaks the composition arcs whenever a saved scene was reopened.

So we decided to switch to Bifrost, since it solved both issues. The problem here is the performance in converting.
A 2.78 milion vertexes maya hierarchy takes less then a minute to convert to usd with maya-usd, while it takes about 10 minutes with bifrost.
Our bifrost converter could still be optimized, but I'm pretty confident from my tests that we could never reach the same performance.

So here's the two questions:
1. Is there a way to remove the 2GB limit? This could solve a big issue and we could let an asset resolver solve the other problem
2.  If I create a layer via Sdf.Layer.CreateAnonymous() is there a way to make it visible to maya's viewport or maya hydra delegate? (maybe via UFE?)

**Comment by vinRosso** [comment]
@santosd @santosg87 Sorry to ping you guys, but maybe you developer have more insights about this?

**Comment by pierrebai-adsk** [comment]
Currently, MayaUSD only handles layers and stage that are inside a MayaUsdProxyNode. If you create layers and stages outside of that node, it won't show up in the viewport.

For the VP2 renderer, we use a VP2 scene delegate, which does the translation between USD geometries and material to Maya VP2 equivalent. It only handles the USD stages it knows about from the MayaUsdProxyNode instances. OTOH, stages and layers can be shared in USD, so it *could* be possible to insert your hand-created layers into a proxy node and it would show up in the viewport. You could consider that "our way" of doing it, even if it was not a direct design.

The 2GB limit is a limit of Maya regarding attribute data size and cannot be worked around. You would need to save layers to disk instead to avoid the limitation.

**Reply by vinRosso** [reply]
Thanks for your answer.

We are currently doing it the other way around you mentioned, by innesting our own usd layers inside the MayaUsdProxy created stage.

The solution I had in mind was to save the SdfLayer I created as a pickled object inside the fileInfo data, so to bypass the 2GB limit.
In that case, could I force your scene delegate to manage my object too?

**Reply by pierrebai-adsk** [reply]
If your layer are inserted in a proxy node, they should show up, since as far as MayaUsd is concerned, they are just normal USD layers.
